cranelift/codegen/src/ir/pcc.rs

```diff
@@ -310,7 +313,10 @@ impl FactContext {
                 if *bit_width < 64 && max > max_value_for_width(width) {
                     return Some(minimal_fact);
                 }
-                Some(minimal_fact)
+                Some(Fact::ValueMax {
+                    bit_width: *bit_width,
+                    max,
+                })
```

Comment:
This was an imprecision in the initial impl -- we had returned the "full range" fact (correct but not useful) rather than the thing we actually computed.